src/hotspot/share/runtime/arguments.hpp

```diff
@@ -269,6 +269,27 @@ class Arguments : AllStatic {
     ExternalProperty
   };
 
+  // RAII style wrapper representing preprocessed arguments returned from Arguments::preprocess and
+  // accepted by Arguments::parse.
+  class Preprocessed final : public StackObj {
+   private:
+    friend class Arguments;
+
+    void* _impl = nullptr;
```

Comment:
I suggest making PreprocessedArguments a private nested class, with only declaration here.  Then make the type of `_impl` be `PreprocessedArguments*` and get rid of a bunch of `static_cast`s in the implementation.  Also give `_impl` a better name.

Reply:
Renamed `PreprocessedArguments` to `PreprocessedImpl`, predeclared in header, got rid of `void*`, and added a comment to `_impl`.